File: Back-end/src/main/java/KJX/KJX/controller/ScrimController.java
package KJX.KJX.controller;

import KJX.KJX.entity.Scrim;
import KJX.KJX.service.ScrimService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/scrims")
@CrossOrigin(origins = "*")
public class ScrimController {
    
    @Autowired
    private ScrimService scrimService;
    
    @GetMapping
    public ResponseEntity<List<Scrim>> getAllScrims() {
        return ResponseEntity.ok(scrimService.getAllScrims());
    }
    
    @GetMapping("/{id}")
    public ResponseEntity<?> getScrimById(@PathVariable Long id) {
        return scrimService.getScrimById(id)
            .map(ResponseEntity::ok)
            .orElseGet(() -> ResponseEntity.notFound().build());
    }
    
    @GetMapping("/status/{status}")
    public ResponseEntity<List<Scrim>> getScrimsByStatus(@PathVariable String status) {
        return ResponseEntity.ok(scrimService.getScrimsByStatus(status));
    }
    
    @GetMapping("/team/{teamId}")
    public ResponseEntity<List<Scrim>> getScrimsByTeamId(@PathVariable Long teamId) {
        return ResponseEntity.ok(scrimService.getScrimsByTeamId(teamId));
    }
    
    @PostMapping
    public ResponseEntity<Scrim> createScrim(@RequestBody Scrim scrim) {
        return ResponseEntity.ok(scrimService.saveScrim(scrim));
    }
    
    @PutMapping("/{id}")
    public ResponseEntity<Scrim> updateScrim(@PathVariable Long id, @RequestBody Scrim scrimDetails) {
        return ResponseEntity.ok(scrimService.updateScrim(id, scrimDetails));
    }
    
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteScrim(@PathVariable Long id) {
        scrimService.deleteScrim(id);
        return ResponseEntity.noContent().build();
    }
}

File: nodejs-backend/src/controllers/scrim.controller.js
/**
 * Scrim Controller
 * Handles scrim management endpoints
 * Converted from Spring Boot ScrimController
 */

const scrimService = require('../services/scrim.service');

class ScrimController {
  /**
   * GET /api/scrims
   * Get all scrims
   */
  async getAllScrims(req, res, next) {
    try {
      const scrims = await scrimService.getAllScrims();
      res.json(scrims);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/scrims/:id
   * Get scrim by ID
   */
  async getScrimById(req, res, next) {
    try {
      const scrim = await scrimService.getScrimById(req.params.id);
      if (!scrim) {
        return res.status(404).json({
          success: false,
          message: 'Scrim not found'
        });
      }
      res.json(scrim);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/scrims/status/:status
   * Get scrims by status
   */
  async getScrimsByStatus(req, res, next) {
    try {
      const scrims = await scrimService.getScrimsByStatus(req.params.status);
      res.json(scrims);
    } catch (error) {
      next(error);
    }
  }

  /**
   * GET /api/scrims/team/:teamId
   * Get scrims by team ID
   */
  async getScrimsByTeamId(req, res, next) {
    try {
      const scrims = await scrimService.getScrimsByTeamId(req.params.teamId);
      res.json(scrims);
    } catch (error) {
      next(error);
    }
  }

  /**
   * POST /api/scrims
   * Create new scrim
   */
  async createScrim(req, res, next) {
    try {
      const scrim = await scrimService.createScrim(req.body);
      res.json(scrim);
    } catch (error) {
      next(error);
    }
  }

  /**
   * PUT /api/scrims/:id
   * Update scrim
   */
  async updateScrim(req, res, next) {
    try {
      const scrim = await scrimService.updateScrim(req.params.id, req.body);
      res.json(scrim);
    } catch (error) {
      if (error.message === 'Scrim not found') {
        return res.status(404).json({
          success: false,
          message: error.message
        });
      }
      next(error);
    }
  }

  /**
   * DELETE /api/scrims/:id
   * Delete scrim
   */
  async deleteScrim(req, res, next) {
    try {
      await scrimService.deleteScrim(req.params.id);
      res.status(204).send();
    } catch (error) {
      if (error.message === 'Scrim not found') {
        return res.status(404).json({
          success: false,
          message: error.message
        });
      }
      next(error);
    }
  }
}

module.exports = new ScrimController();
